file_writer.go

```diff
@@ -176,6 +201,28 @@ func (f *FileWriter) SetDeadline(t time.Time) error {
 // of this, it is important that Close is called after all data has been
 // written.
 func (f *FileWriter) Write(b []byte) (int, error) {
+	// Encrypt data chunk if file in HDFS encrypted zone.
+	if f.enc != nil && len(b) > 0 {
+		var offset int
+		for offset < len(b) {
+			size := min(len(b)-offset, aesChunkSize)
+			ciphertext, err := aesCtrStep(f.offset, f.enc, b[offset:offset+size])
+			if err != nil {
+				return offset, err
+			}
+			writtenSize, err := f.writeImpl(ciphertext)
```

Comment:
What happens if a aes chunk gets split across multiple blocks? I guess it's fine?

We could test this by setting blocksize to 16kb with `CreateFile` and calling `Write` with 1mb of data.

Reply:
I've add test `TestEncryptedZoneLargeBlock` with hdfs block 128Kb and 1.5Mb of data. It passed.

Reply:
That test is helpful, but it actually just tests that our write and read code expects the same format. Ideally we would write with go and read with java to actually test that behavior.

Reply:
Sure! I've add checks via 'hadoop dfs' too.